android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java

```diff
@@ -172,33 +174,37 @@ public void sendToNotificationCentre(Bundle bundle) {
             String bundle_title = bundle.getString("bundle_title");
             String bundle_id = bundle.getString("bundle_id");
 
-            NotificationCompat.Builder notificationBuilder =
-                    new NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID)
-                            .setSmallIcon(smallIconResId)
-                            .setContentTitle(title)
+            NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context,
+                    NOTIFICATION_CHANNEL_ID).setSmallIcon(smallIconResId).setContentTitle(title)
                             .setContentText(message);
 
-
             NotificationManager notificationManager = notificationManager();
             checkOrCreateChannel(notificationManager);
 
-            if(bundle_title != null && bundle_id != null) {
-                //LP: is supposed to be grouped message
+            Intent intent = new Intent(context, intentClass);
+            intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
+            bundle.putBoolean("userInteraction", true);
+            intent.putExtra("notification", bundle);
```

Comment:
is this extra field correct? Or this works only if the backend sends the "notification" bit (instead of `data`) ?

Reply:
In the RN client where the package is being used, we have this:
```
PushNotification.configure({
    onRegister,
    onNotification: (notification: any) =>
      handlePushNotifications(notification),
    senderID: gcm,
    permissions: {
      alert: true,
      badge: true,
      sound: true,
    },
    popInitialNotification: true,
    requestPermissions: true,
  });
```

So I think it's correct to have this like this.